src/compiler/parser.ts

```diff
@@ -5558,9 +5559,9 @@ namespace Parser {
         );
     }
 
-    function parseBinaryExpressionOrHigher(precedence: OperatorPrecedence): Expression {
+    function parseBinaryExpressionOrHigher(precedence: OperatorPrecedence, previousNode?: Node): Expression {

```

Comment:
From https://github.com/microsoft/TypeScript/pull/56101#issuecomment-1819801432:

> 
I'm not totally sure how I feel about threading the previous node down into the parser like this, honestly. I'm not sure of a better way, though, besides passing down a boolean that's true if it's a type assertion or something.

I'll just go ahead and say this: I hate it. Adding in the first concept of a "previous" node to a parser for a specialized error message makes me unhappy.

An alternative could be to keep a variable alongside `var currentToken: SyntaxKind` & co. I dislike that ... more, honestly. But it's an option. 🤷